File: src/Domain/Expenses/Expense.cs
namespace Domain.Expenses;

public sealed class Expense
{
    public Guid Id { get; set; }
    public Guid UserId { get; set; }
    public Guid AccountId { get; set; }
    public decimal Amount { get; set; }
    public string? Note { get; set; }
    public string? Category { get; set; }
    public DateOnly Date { get; set; }
    public DateTime CreatedAtUtc { get; set; }
}

File: src/Web.Client/lib/types/expense.ts
export type Expense = {
  id: string;
  accountId: string;
  accountName: string;
  amount: number;
  note: string | null;
  category: string | null;
  date: string;
};

export type ExpenseRequest = {
  accountId: string;
  amount: number;
  note: string | null;
  category: string | null;
  date: string;
};
